lib/rooms/stats.ex

```diff
@@ -52,3 +52,31 @@ defmodule SignalTower.Stats do
     end
   end
 end
+
+defmodule SignalTower.PrometheusStats do
+  use Prometheus.Metric
+  @counter [name: :palava_joined_room_total,
+            labels: [],
+            help: "Number of peers joined a room"]
+
+  @counter [name: :palava_leave_room_total,
+            labels: [],
+            help: "Number of peers left a room"]
+
+  def reset() do
+    Counter.reset(name: :palava_joined_room_total)
+    Counter.reset(name: :palava_leave_room_total)
+  end
+
+  def join() do
+    Counter.inc(name: :palava_joined_room_total)
+  end
+
+  def leave() do
+    Counter.inc(name: :palava_leave_room_total)
+  end
+
+  def to_string() do
+    Prometheus.Format.Text.format
```

Comment:
Does this return always the whole history of the counter? If so that's bad isn't it?

Reply:
Well, I guess for a counter it only shows the current value, right? But what if we want to add gauge metrics? How do we offer only the metrics that have not yet been collected by prometheus? (or is that not how it should work)

Reply:
> Does this return always the whole history of the counter? If so that's bad isn't it?

It always show the current value. For all kind of prometheus metrics. The prometheus time series database polls these metrics periodically to get the history.

> But what if we want to add gauge metrics

For the the current metrics you can calculate joined peers - left peers. You can also use a gauge metrics type which can be incremented and decremented, but its better to collect the raw data and do the processing later if possible.

> How do we offer only the metrics that have not yet been collected by prometheus? (or is that not how it should work)

You mean the diff to the last collected state? I don't understand your question. Maybe you can give me an example of a particular metric you have in mind.

Reply:
Ok, the counter is a bad example here, because it does not matter if we miss any updates. Let's take as example the session duration, will to_string() show only the latest entry then? Wouldn't this lead to many missed entries, if between two prometheus polls, several sessions end and thus several log entries get added?

Reply:
`to_string()` just gives you a snapshot of the signal tower at the moment the method is called. So you are right that you might miss updates when you poll just every 15 seconds (what is default for prometheus). But in the most cases it does not matter.

- For counter it does not matter at all because their value only increases so you might poll when the value is 0 and then poll again when the value is 10
- For gauges you might miss maximum and minimum because you might poll when the value is 5 and then poll again when the value is 10, but in between it was 2.
- For histograms you only loose precision because you usually choose your buckets not that fine granular. 

For session duration I would choose a histogram. Thus you get a session duration histogram over time.